### PR TITLE
usb_smartcard_sharing: Add check for spice connection

### PR DESCRIPTION
**Pull request by ybduan**
usb smartcard sharing works only with spice.

ID: 1964826

Signed-off-by: ybduan <yduan@redhat.com>

### CODE REVIEW

**Comment by ybduan** [comment]
(1/1) Host_RHEL.m8.ux.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.x.0.x86_64.io-github-autotest-qemu.usb.usb_smartcard_sharing.without_usb_hub.qemu-xhci.q35: CANCEL: Only support spice connection (30.65 s)

**Comment by ybduan** [comment]
Please help review this PR, thanks! @yanan-fu

**Comment by yanan-fu** [comment]
I think `use_redir` have the same issue, which use chardev_backend  `spicevmc` and `display = spice` too.

**Comment by ybduan** [comment]
Close the PR for now, and will consider the details later.

**Comment by ybduan** [comment]
Please help review this PR, thanks! @yanan-fu

File: qemu/tests/usb_redir.py [hunk 1]
```diff
@@ -233,6 +233,9 @@ def _usb_stick_io(mount_point):
     usb_stick = "Mass Storage" in process.getoutput(lsusb_cmd)
     rv_binary = params.get('rv_binary', 'remote-viewer')
 
+    if backend == 'spicevmc' and params.get('display') != 'spice':
+        test.cancel("Only support spice connection")
+
```
Comment (yanan-fu) [inline qemu/tests/usb_redir.py:238]: Better move to L227, cancel/error test after get backend.

File: qemu/tests/usb_smartcard_sharing.py [hunk 2]
```diff
@@ -145,6 +145,9 @@ def _rv_connection_check():
     sc_db_dst = params.get('sc_db_dst', '/home/fake_smartcard')
     ccid_info = params.get('ccid_info', 'Gemalto')
 
+    if params.get('display') != 'spice':
+        test.cancel("Only support spice connection")
+
```
Comment (yanan-fu) [inline qemu/tests/usb_smartcard_sharing.py:150]: Same,  move to L141 is better.

**Comment by ybduan** [comment]
Updated, thanks!

**Comment by yanan-fu** [comment]
Call for test result (no effect for other test), thanks!

**Comment by ybduan** [comment]
python3 ConfigTest.py --testcase=usb --platform=x86_64 --guestname=RHEL.8.5.0 --machines=q35 --clone=no
......
Summary: 
Finshed=205, PASS=187, SKIP=18

**Comment by yanan-fu** [comment]
Ack

**Pull request merged**